agent/src/com/thoughtworks/go/agent/AgentController.java

```diff
@@ -62,13 +64,17 @@
     private SubprocessLogger subprocessLogger;
     private final SystemEnvironment systemEnvironment;
     private AgentUpgradeService agentUpgradeService;
+    private PackageAsRepositoryExtension packageAsRepositoryExtension;
+    private SCMExtension scmExtension;
     private TaskExtension taskExtension;
 
     @Autowired
     public AgentController(BuildRepositoryRemote server, GoArtifactsManipulator manipulator, SslInfrastructureService sslInfrastructureService, AgentRegistry agentRegistry,
                            AgentUpgradeService agentUpgradeService, SubprocessLogger subprocessLogger, SystemEnvironment systemEnvironment,
-                           PluginManager pluginManager, TaskExtension taskExtension) {
+                           PluginManager pluginManager, PackageAsRepositoryExtension packageAsRepositoryExtension, SCMExtension scmExtension, TaskExtension taskExtension) {
```

Comment:
Should we have ```Extensions``` or something? I don't see the need to change this every time we add a new kind of extension. Could use ```extensions.forTask()``` and ```extensions.forPackageRepository()```, etc.